packages/front-end/components/Settings/EditDataSource/ExperimentAssignmentQueries/AddEditExperimentAssignmentQueryModal.tsx

```diff
@@ -124,35 +187,54 @@ export const AddEditExperimentAssignmentQueryModal: FC<EditExperimentAssignmentQ
               required
               {...form.register("userIdType")}
             />
-
+            <StringArrayField
+              label="Dimension Columns"
+              value={userEnteredDimensions}
+              onChange={(dimensions) => {
+                form.setValue("dimensions", dimensions);
+              }}
+            />
             <div className="row">
               <div className="col">
-                <CodeTextArea
+                <Field
```

Comment:
Don't use `Field` here.  It does a bunch of extra things, including adding extra padding below and setting accessibility attributes that don't apply when using `render`.  Just use `<div>` and `<label>` directly instead.